File: app/src/main/java/com/example/shopsphere/CleanArchitecture/domain/GetProductsByCategoryUseCase.kt
package com.example.shopsphere.CleanArchitecture.domain

import javax.inject.Inject

class GetProductsByCategoryUseCase  @Inject constructor(private val repository: IRepository) {

    suspend fun getProductsByCategory(category: String): Result<List<DomainProductResult>> {
        return repository.getProductsByCategory(category)
    }
}

File: ShopSphere-iOS/ShopSphere/Domain/UseCases/GetProductsByCategoryUseCase.swift
import Foundation

struct GetProductsByCategoryUseCase {
    private let repository: RepositoryProtocol

    init(repository: RepositoryProtocol) {
        self.repository = repository
    }

    func execute(category: String) async throws -> [Product] {
        try await repository.getProductsByCategory(category)
    }
}
